rewrite-core/src/main/java/org/openrewrite/trait/reference/Reference.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.trait.reference;
+
+import org.openrewrite.*;
+import org.openrewrite.trait.Trait;
+
+import java.util.Set;
+
+@Incubating(since = "8.39.0")
+public interface Reference extends Trait<Tree> {
```

Comment:
I get a warning on the `Tree` type parameter here, as we don't have a `pacakge-info.java` with Jspecify annotations.

Reply:
No longer an issue after moving the reference trait up a package